### ISSUE TITLE
Display fixed row in b-table

### ISSUE THREAD

**Issue by ccamarat** [bootstrap-vue/bootstrap-vue#682]
Suppose you're using `<b-table>` to build a directory browser. The first row is always "Up to parent". If the directory is empty, you want to display "No files" or something.
The "up to parent" problem _could_ be solved (albeit clunkily) by adding a dummy node to the `items` collection, but doing so prevents the collection from ever being empty, so the `empty` slot will never appear. A more ideal solution would be to be able to inject a fixed "template-row" of some kind, but I'm not seeing anything like this in the docs. Any thoughts on how one might tackle this problem?

**Comment by tmorehouse** [comment]
I don't think this is possible with the current implementation of `<b-table>`.

You could take your dummy row one step further and create a second dummy row that has the "No Files" message in it.  And on the first row (up to parent) use a custom format to render a `@click.stop` handler to handle your navigation to parent folder.

The other option might be to dynamically change the `fields` object to be 'Up to Parent' in the first header field, and then use the empty slot to say 'no files found`.

**Comment by ccamarat** [comment]
I think I can work with extrapolating the dummy row example as you suggested. Appreciate the help!

**Comment by tmorehouse** [comment]
Let us know how it goes for you!

**Comment by ccamarat** [comment]
Yea, this approach works. Still a a couple quirks I needed to work through:
* The "nothing to see here" row appears clickable (leveraging `tdClass`)
* The embedded file viewer requires changing the path based on type

I can live with the first and the second can be addressed by hacking around it with more attrs on the model / intercepting click handlers. I can improve the abstraction of some of this with a better data model, but it feels like it would be better to be able to address in the view.

What would you think about adding support for something like an optional `<b-row-template>` child to `b-table`? It could provide greater semantic flexibility when working with different parts of the table, plus we could use Vue's native [`<component>`](https://v1.vuejs.org/guide/components.html#Template-Parsing) element to dynamically change templates. In this example one could use it like:
```
<template>
  <b-table :items="children">
    <component :is="getComponent" :scope="child"></component>
  </b-table>
</template>

<script>
export default {
  props: ['children'],
  methods: {
    getComponent (item) {
      return `${item.type}-row`;
    }
};
</script>
```

The innards of `b-row-template` would support templates the same way `<b-table>` innards currently do.

**Comment by tmorehouse** [comment]
Hmmm that is an interesting concept.

**Comment by tmorehouse** [comment]
Another way might be to add a two new slots, bot scoped, and both being a TR (one for top, and one for bottom).  and then passing the fields object as a scoped variable. It would be up to the user to render the correct number of TD cells (or spanned TDs).  

Potentially, we could pass the number of TD cells required, an array of field names (keys of the field object), and then the fields object (although technically you would have access to this already as you passed this to b-table to begin with).

**Comment by tmorehouse** [comment]
The above PR would let you do the following:

```html
<b-table ...>
  <template slot="top-row" scope="data">
    <td v-for="idx in data.columns">Fixed Cell {{ idx }}</td>
  </template>
  <template slot="bottom-row" scope="data">
    <td :span="data.columns">Something special here</td>
  </template>
</b-table>
```

Bottom row is always rendered below the empty slots

Proviedd scoped data data is `columns` the number of TD (or TH) cells that should be rendered, and `fields` which is the original fields object

**Comment by tmorehouse** [comment]
Should now be available in release v1.0.0.beta.6
